### PR TITLE
update image for newer versions

### PR DESCRIPTION
**Pull request by UltimateDogg**
base image off fedora 30, pip 19. install robot framework db, ssh and
ftp libs. for pip use --no-cache-dir to save space in image

### CODE REVIEW

**Comment by UltimateDogg** [comment]
Maybe it would be good to tag a release docker before accepting this too

File: Dockerfile [hunk 1]
```diff
@@ -21,11 +21,14 @@ ENV FAKER_VERSION 4.2.0
...
 ENV PABOT_VERSION 0.53
-ENV PYTHON_PIP_VERSION 18.0*
+ENV PYTHON_PIP_VERSION 19.0*
 ENV REQUESTS_VERSION 0.5.0
 ENV ROBOT_FRAMEWORK_VERSION 3.1.1
 ENV SELENIUM_LIBRARY_VERSION 3.3.1
 ENV XVFB_VERSION 1.20.*
+ENV DATABASE_LIBRARY_VERSION 1.2
+ENV SSH_LIBRARY_VERSION 3.3.0
+ENV FTP_LIBRARY_VERSION 1.6
```
Comment (ppodgorsek) [inline Dockerfile]: Could you keep them sorted alphabetically please?

File: Dockerfile [hunk 2]
```diff
@@ -38,15 +41,21 @@ RUN dnf upgrade -y \
...
+  --no-cache-dir \
   robotframework==$ROBOT_FRAMEWORK_VERSION \
   robotframework-faker==$FAKER_VERSION \
   robotframework-pabot==$PABOT_VERSION \
   robotframework-requests==$REQUESTS_VERSION \
-  robotframework-seleniumlibrary==$SELENIUM_LIBRARY_VERSION
+  robotframework-seleniumlibrary==$SELENIUM_LIBRARY_VERSION \
+  robotframework-databaselibrary==$DATABASE_LIBRARY_VERSION \
+  robotframework-sshlibrary==$SSH_LIBRARY_VERSION \
+  robotframework-ftplibrary==$FTP_LIBRARY_VERSION
```
Comment (ppodgorsek) [inline Dockerfile]: Could you keep them sorted alphabetically please?

File: Dockerfile [hunk 3]
```diff
@@ -61,8 +70,7 @@ COPY bin/chromium-browser.sh /opt/robotframework/bin/chromium-browser
 COPY bin/run-tests-in-virtual-screen.sh /opt/robotframework/bin/
 
 # FIXME: below is a workaround, as the path is ignored
-RUN mv /usr/lib64/chromium-browser/chromium-browser /usr/lib64/chromium-browser/chromium-browser-original \
-  && ln -sfv /opt/robotframework/bin/chromium-browser /usr/lib64/chromium-browser/chromium-browser
+RUN ln -sfv /opt/robotframework/bin/chromium-browser /usr/lib64/chromium-browser/chromium-browser
```
[the remark this answers is not in the document]
Comment (UltimateDogg) [reply Dockerfile]: I agree its a bit easier to understand your way, but doing it this way only has one layer with chrome in it. Yours has chrome in 2 layers making the image bigger by the size of chrome (about 40mb).

**Comment by UltimateDogg** [comment]
sure, updated per comments

File: Dockerfile [hunk 4]
```diff
@@ -12,19 +12,22 @@ ENV SCREEN_HEIGHT 1080
 ENV SCREEN_WIDTH 1920
 
 # Set number of threads for parallel execution
-# By default, no parallelisation
+# By default, no parallelization
```
Comment (ppodgorsek) [inline Dockerfile]: This was not a mistake, it is British English, please revert.

File: Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (ppodgorsek) [reply Dockerfile]: In that case, it doesn't make sense to keep `ln` here, please move it with the `mv` command, the COPY lines should also be moved between the "Dependency versions" and "Install system dependencies" sections.

**Comment by ppodgorsek** [comment]
Several tickets were created to track the changes in this pull request, in order to help with the release of the next version:
* #164 
* #167 
* #168 
* #169 
* #170 
* #171

File: Dockerfile [hunk 5]
```diff
@@ -38,15 +46,22 @@ RUN dnf upgrade -y \
     xorg-x11-server-Xvfb-$XVFB_VERSION \
     which \
     wget \
-  && dnf clean all
+  && dnf clean all \
+  && mv /usr/lib64/chromium-browser/chromium-browser /usr/lib64/chromium-browser/chromium-browser-original
```
Comment (ppodgorsek) [inline Dockerfile]: Missing `\` at the end of the line

File: Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (UltimateDogg) [reply Dockerfile]: done

File: Dockerfile (hunk 4 above)
Comment (UltimateDogg) [reply Dockerfile]: done

**Comment by UltimateDogg** [comment]
> 
> 
> All good, thank you very much! :) I'll merge it as soon as the build passes.

no problem, thanks for the image

**Pull request merged**